### ISSUE TITLE
Go through CL/server tutorials, report/fix issues

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#1978]
There are 3 tutorials under https://dart.dev/tutorials/server. We'd like a volunteer for each one, to go through and report/fix issues (similar to #1958).

* [x] [Get started](https://dart.dev/tutorials/server/get-started) (reviewer @foxbit19 found no issues)
* [x] [Write command-line apps](https://dart.dev/tutorials/server/cmdline) (reviewer @wang-ories found and fixed 2 issues)
* [x] [Write HTTP clients & servers](https://dart.dev/tutorials/server/httpserver) (reviewer: @hamidrezabstn)

Known issue:
* #432 (no mention of pubspec in https://dart.dev/tutorials/server/cmdline)

**Comment by wang-ories** [comment]
Hello @kwalrath  I can work on this  tutorial.

**Comment by kwalrath** [comment]
Thanks, @wang-ories! Which one would you like to work on first?

**Comment by kwalrath** [comment]
@wang-ories you get your pick... just tell us which one you're doing, so anyone else who's interested can sign up for another tutorial.

**Comment by wang-ories** [comment]
@kwalrath  I take  this tutorial : **Write command-line apps**

**Comment by wang-ories** [comment]
@kwalrath  I found two smalls problem in this tutorial:

1 - In this section https://dart.dev/tutorials/server/cmdline#overview-of-the-dcat-app-code : the result of this command   **$ dart dcat.dart -n quote.txt**  produces empty output, i.e the file quote.txt must be created first. In the tutorial the output is  three quotes from **quote.txt.**

2 - This link https://api.dart.dev/stable/dart-io/dart-convert-library.html at the end of the page can’t be found.

**Comment by kwalrath** [comment]
> 1 - In this section https://dart.dev/tutorials/server/cmdline#overview-of-the-dcat-app-code : the result of this command **$ dart dcat.dart -n quote.txt** produces empty output, i.e the file quote.txt must be created first. In the tutorial the output is three quotes from **quote.txt.**

Ah right, back when this was written, there was an in-page link to all the files, including quote.txt. The source is at https://github.com/dart-lang/site-www/blob/master/examples/misc/test_data/quote.txt. Maybe we should just tell people to download it from that location.

> 2 - This link https://api.dart.dev/stable/dart-io/dart-convert-library.html at the end of the page can’t be found.

Looks like it should be https://api.dart.dev/stable/dart-convert/dart-convert-library.html.

**Comment by wang-ories** [comment]
Ok

**Comment by wang-ories** [comment]
@kwalrath  for the link is simple, can  I just fix it?

**Comment by kwalrath** [comment]
> @kwalrath for the link is simple, can I just fix it?

Yep, go ahead.

**Comment by wang-ories** [comment]
> > 1 - In this section https://dart.dev/tutorials/server/cmdline#overview-of-the-dcat-app-code : the result of this command **$ dart dcat.dart -n quote.txt** produces empty output, i.e the file quote.txt must be created first. In the tutorial the output is three quotes from **quote.txt.**
> 
> Ah right, back when this was written, there was an in-page link to all the files, including quote.txt. The source is at https://github.com/dart-lang/site-www/blob/master/examples/misc/test_data/quote.txt. Maybe we should just tell people to download it from that location.
> 
> > 2 - This link https://api.dart.dev/stable/dart-io/dart-convert-library.html at the end of the page can’t be found.
> 
> Looks like it should be https://api.dart.dev/stable/dart-convert/dart-convert-library.html.

@kwalrath , I can work on this first problem, if not assigned yet. I will add the link to download and tell people to download.

**Comment by kwalrath** [comment]
> @kwalrath , I can work on this first problem, if not assigned yet. I will add the link to download and tell people to download.

That would be great. Thank you!

**Comment by wang-ories** [comment]
> > @kwalrath , I can work on this first problem, if not assigned yet. I will add the link to download and tell people to download.
> 
> That would be great. Thank you!

Ok, I made the changes in a new PR.

**Comment by foxbit19** [comment]
Hi @kwalrath, 
I can take **Get started** tutorial!

**Comment by kwalrath** [comment]
> I can take **Get started** tutorial!
It's yours!

**Comment by foxbit19** [comment]
I've checked all commands and links into **Get started** tutorial. Everything seems to work as expected!

**Comment by hamidrezabstn** [comment]
hi can i pick third one ?

**Comment by hamidrezabstn** [comment]
:) :?

**Comment by kwalrath** [comment]
@hamidrezabstn the third one is yours. Thanks!

**Comment by hamidrezabstn** [comment]
@kwalrath
should i send PR or report issues here ?

**Comment by hamidrezabstn** [comment]
**A quick glance at the code**

code example is different from github source code
_Launch the web server_ -> in this section user need more info about webdev and installing it then he/she could run it.

**Making a POST request from a standalone client**

client code sample is differ from github version

**Using the http_server package**

mini code sample is differ from github version
and beside web site code is working but github sample is not working
basic code sample is differ from github version
static code sample is differ from github version

**Using https with bindSecure()**

secure code sample is differ from github version

**all links are fine**

**Comment by kwalrath** [comment]
Thanks very much, @hamidrezabstn. Could you please file one issue for the code sample differences and an issue or PR for the webdev issue? Thanks!

**Comment by hamidrezabstn** [comment]
@kwalrath 
Your welcome! 
i will make changes and send PR

**Comment by hamidrezabstn** [comment]
i made a PR at #2021

**Comment by kwalrath** [comment]
@hamidrezabstn thanks for fixing the sample links! Do you want to take a stab at fixing the webdev issue, or do you want to file an issue about it?

**Comment by hamidrezabstn** [comment]
@kwalrath i'm really sorry about that i totally forget about it ! i will make a PR for it!

**Comment by hamidrezabstn** [comment]
@kwalrath Done! #2025

**Comment by kwalrath** [comment]
@hamidrezabstn I just noticed that the names of the source files are down below the snippets. E.g. _hello_world_server.dart_ is displayed beneath the first code excerpt (https://dart.dev/tutorials/server/httpserver#a-quick-glance-at-the-code). It looks odd.

(I tried looking at it in the wayback machine, but it's having issues with that page. I think the filename used to be displayed on top of the code sample, but I'm thinking that isn't really necessary.)

One solution would be to simply remove the lines that have these filenames. E.g. for the first example we might remove this line:

```
<div class="prettify-filename">hello_world_server.dart</div>
```

What do you think?

**Comment by chalin** [comment]
Maybe during site migration some styles were dropped or overridden. I can take a look first before we strip out filenames. WDYT @kwalrath?

**Comment by kwalrath** [comment]
> Maybe during site migration some styles were dropped or overridden. I can take a look first before we strip out filenames. WDYT @kwalrath?

Sure, that'd be great @chalin.

**Comment by kwalrath** [comment]
@chalin fixed the filename appearance in #2026, so I think we're good to close this issue. Thank you, everyone!!

**Comment by hamidrezabstn** [comment]
@kwalrath i think because of file links under each section's title , for example: [Example file for this section: hello_world_server.dart.](https://dart.dev/tutorials/server/httpserver#run-hello-world) , and referencing that file in different positions in related section, like this

> At the command line, run the hello_world_server.dart script:

,that file name below snippets is not necessary .
but if it should be there i think below is better than above. its less disturbing.
